### DISCUSSION TITLE
How to call a bound method object in a dynamic C module?

### DISCUSSION THREAD

**Discussion by TsXor** [micropython/micropython#9882] [Using MicroPython]
`mp_call_function_n_kw` causes `LoadProhibited`
`bound_meth_call` is not available in dynruntime.h
`mp_call_method_n_kw` cannot compile: `undefined symbol: mp_call_method_n_kw`

note: lvgl version

**Comment by jimmo** [comment]
> `mp_call_function_n_kw` causes `LoadProhibited`

This is the way to do it. You shouldn't care if the method is bound or not, you just call it like a regular method.

I suspect something is wrong with the method pointer or the way it's being created. Where does it come from -- is it passed to your code from Python or does the C code bind it?

**Reply by TsXor** [reply]
My mistake, I allocated a struct on stack and passed its pointer somewhere else...
However, after I used `m_malloc` to allocate it on heap, it causes `Interrupt wdt timeout on CPU0`

**Reply by TsXor** [reply]
My mistake again, I tried to modify the data of an object(`obj_ptr->data`) at C side.
